### PR TITLE
Fixing byte array construction

### PR DESCRIPTION
**Pull request by GhostofCookie**
And adding the ability to left align when constructing from a byte array.

Also, general cleanup.

### CODE REVIEW

File: include/quicr/hex_endec.h [hunk 1]
```diff
@@ -65,7 +65,8 @@ class HexEndec
     template<Unsigned... UInt_ts>
     static inline std::string Encode(std::span<uint16_t> distribution, UInt_ts... values)
     {
-        static_assert(Size >= (Dist + ...), "Total bits cannot exceed specified size");
+        if(Size < std::accumulate(distribution.begin(), distribution.end(), 0))
```
Comment (TimEvens) [inline include/quicr/hex_endec.h:68]: So dist of bits doesn't have to total size?

File: include/quicr/name.h [hunk 2]
```diff
@@ -152,58 +156,77 @@ class Name
...
         else
         {
             _hi = 0;
-            _lo = hex_to_uint<uint_type>(hex_value.substr(0, hex_value.length()));
+            _lo = hex_to_uint<uint_t>(hex_value.substr(0, hex_value.length()));
         }
-    }
 
-    constexpr Name& operator=(std::string_view hex) { return *this = Name(hex); }
+        if (align_left) *this <<= (sizeof(Name) - (hex_value.length() / 2)) * 8;
```
Comment (TimEvens) [inline include/quicr/name.h:168]: why divide by 2 here?

**Review by TimEvens — APPROVED** [review]
LGTM

File: include/quicr/hex_endec.h (hunk 1 above)
Comment (GhostofCookie) [reply include/quicr/hex_endec.h:68]: No, you're allowed to let the rest of it be filled with zeros. The only constraint in this Encode is that you can't exceed Size. The remaining bits will be 0s (numero-uri uses this functionality).

File: include/quicr/name.h (hunk 2 above)
Comment (GhostofCookie) [reply include/quicr/name.h:168]: Because a hex string will have 2 characters for each byte

**Pull request merged**